spring-integration-core/src/main/java/org/springframework/integration/dsl/ConsumerEndpointSpec.java

```diff
@@ -170,15 +173,19 @@ public S requiresReply(boolean requiresReply) {
 	/**
 	 * @param sendTimeout the send timeout.
 	 * @return the endpoint spec.
-	 * @see AbstractReplyProducingMessageHandler#setSendTimeout(long)
+	 * @see AbstractMessageProducingHandler#setSendTimeout(long)
 	 */
 	public S sendTimeout(long sendTimeout) {
 		assertHandler();
-		if (this.handler instanceof AbstractReplyProducingMessageHandler) {
-			((AbstractReplyProducingMessageHandler) this.handler).setSendTimeout(sendTimeout);
+		if (this.handler instanceof AbstractMessageProducingHandler) {
+			((AbstractMessageProducingHandler) this.handler).setSendTimeout(sendTimeout);
+		}
+		else if (this.handler instanceof AbstractMessageRouter) {
+			// This should probably go on the RouterSpec, but we put it here for consistency
```

Comment:
Yes, it's the right decision, my question was really why we decided to put it on the consumer spec in the first place, given that it's a handler property. But, I agree, it's too late to change now.